src/api/routes/bns/names.ts

```diff
@@ -0,0 +1,101 @@
+
+export function createBNSNamesRouter(db: DataStore): RouterWithAsync {
+  const router = addAsync(express.Router());
+
+  router.getAsync('/:name/zonefile/:zoneFileHash', async (req, res) => {
+    // Fetches the historical zonefile specified by the username and zone hash.
+    const { name, zoneFileHash } = req.params;
+
+    const nameQuery = await db.getName({ name: name });
+    if (nameQuery.found) {
+      const zonefile = await db.getHistoricalZoneFile({ name: name, zoneFileHash: zoneFileHash });
+      if (zonefile.found) {
+        res.json(zonefile.result);
+      } else {
+        res.status(404).json({ error: 'No such zonefile' });
+      }
+    } else {
+      res.status(400).json({ error: 'Invalid name or subdomain' });
+    }
+  });
+
+  router.getAsync('/:name/zonefile', async (req, res) => {
+    // Fetch a user’s raw zone file. This only works for RFC-compliant zone files. This method returns an error for names that have non-standard zone files.
+    const { name } = req.params;
+
+    const nameQuery = await db.getName({ name: name });
+    if (nameQuery.found) {
+      const zonefile = await db.getLatestZoneFile({ name: name });
+      if (zonefile.found) {
+        res.json(zonefile.result);
+      } else {
+        res.status(404).json({ error: 'No zone file for name' });
+      }
+    } else {
+      res.status(400).json({ error: 'Invalid name or subdomain' });
+    }
+  });
+
+  router.getAsync('/', async (req, res) => {
+    const page = parsePagingQueryInput(req.query.page ?? 0);
+
+    const { results } = await db.getNamesList({ page });
+    if (results.length === 0 && req.query.page) {
+      res.status(400).json(BNSErrors.InvalidPageNumber);
+    }
+    res.json(results);
+  });
+
+  router.getAsync('/:name', async (req, res) => {
+    const { name } = req.params;
+    let nameInfoResponse: BNSGetNameInfoResponse;
+    // Subdomain case
+    if (name.split('.').length == 3) {
+      const subdomainQuery = await db.getSubdomain({ subdomain: name });
+      if (!subdomainQuery.found) {
+        return res.status(404).json({ error: `cannot find subdomain ${name}` });
+      }
+      const { result } = subdomainQuery;
+
+      nameInfoResponse = {
+        address: result.owner,
+        blockchain: bnsBlockchain,
+        last_txid: result.tx_id ? result.tx_id : '',
+        resolver: result.resolver,
+        status: 'registered_subdomain',
+        zonefile: result.zonefile,
+        zonefile_hash: result.zonefile_hash,
+      };
+    } else {
+      const nameQuery = await db.getName({ name });
+      if (!nameQuery.found) {
+        return res.status(404).json({ error: `cannot find name ${name}` });
+      }
+      const { result } = nameQuery;
+      nameInfoResponse = {
+        address: result.address,
+        blockchain: bnsBlockchain,
+        expire_block: result.expire_block,
+        grace_period: result.grace_period,
+        last_txid: result.tx_id ? result.tx_id : '',
```

Comment:
The `tx_id` here and above need converted to a string. See https://stacks-node-api-bns.testnet.stacks.co/v1/names/hankstoever.id
```json
{
  "address": "SP2JZCEMXRFJTS74WM99547NC4XX0F1FCAY9T7R7Y",
  "blockchain": "stacks",
  "expire_block": 52595,
  "last_txid": "0x[object Object]",
  "status": "",
  "zonefile": "$ORIGIN hankstoever.id\n$TTL 3600\n_http._tcp\tIN\tURI\t10\t1\t\"https://gaia.blockstack.org/hub/1G8XTwZkUzu7DJYDW4oA4JX5shnW8LcpC2/profile.json\"\n\n",
  "zonefile_hash": "daa4437cd303d5c751b62a4c25ece524889b0b81"
}
```

Reply:
done! The issue was in `postgres-store.ts`